server/src-lib/Hasura/GraphQL/Transport/WebSocket.hs

```diff
@@ -219,6 +224,9 @@ onConn (L.Logger logger) corsPolicy wsId requestHead = do
       _                   ->
         throw404 "only '/v1/graphql', '/v1alpha1/graphql' are supported on websockets"
 
+    assertAcceptingConns ShuttingDown = throw500 "server is shuting down"
+    assertAcceptingConns _ = pure ()
```

Comment:
I’d explicitly pattern match on `AcceptingConns` here rather than using the wildcard pattern to make it more future-proofed against changes to `ServerStatus`.

Reply:
good call